cpp/tests/test_abm_simulation.cpp

```diff
@@ -103,16 +63,75 @@ TEST(TestSimulation, getWorldAndTimeConst)
+    auto school_id   = world.add_location(mio::abm::LocationType::School);
+    auto social_id   = world.add_location(mio::abm::LocationType::SocialEvent);
+    auto basics_id   = world.add_location(mio::abm::LocationType::BasicsShop);
+    auto public_id   = world.add_location(mio::abm::LocationType::PublicTransport);
+
+    auto& person1 = add_test_person(world, home_id, AGE_GROUP_5_TO_14, mio::abm::InfectionState::Exposed);
+    auto& person2 = add_test_person(world, home_id, AGE_GROUP_15_TO_34, mio::abm::InfectionState::Exposed);
+    auto& person3 = add_test_person(world, home_id, AGE_GROUP_35_TO_59, mio::abm::InfectionState::Dead);
+    person1.set_assigned_location(home_id);
+    person2.set_assigned_location(home_id);
+    person3.set_assigned_location(home_id);
+    person1.set_assigned_location(school_id);
+    person2.set_assigned_location(work_id);
+    person2.set_assigned_location(icu_id);
+    person2.set_assigned_location(hospital_id);
+    person1.set_assigned_location(social_id);
+    person2.set_assigned_location(social_id);
+    person3.set_assigned_location(social_id);
+    person1.set_assigned_location(basics_id);
+    person2.set_assigned_location(basics_id);
+    person3.set_assigned_location(basics_id);
+    person2.set_assigned_location(public_id);
+
+    mio::abm::TripList& trip_list = world.get_trip_list();
+
+    mio::abm::Trip trip1(person2.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(2), work_id);
+    mio::abm::Trip trip2(person2.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(3), icu_id);
+    mio::abm::Trip trip3(person2.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(4), hospital_id);
+    mio::abm::Trip trip4(person2.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(5), social_id);
+    mio::abm::Trip trip5(person2.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(6), basics_id);
+    mio::abm::Trip trip6(person2.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(7), public_id);
+    mio::abm::Trip trip7(person2.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(8), home_id);
+
+    trip_list.add_trip(trip1);
+    trip_list.add_trip(trip2);
+    trip_list.add_trip(trip3);
+    trip_list.add_trip(trip4);
+    trip_list.add_trip(trip5);
+    trip_list.add_trip(trip6);
+    trip_list.add_trip(trip7);
+
+    mio::History<mio::DataWriterToMemory, mio::abm::LogLocationInformation, mio::abm::LogPersonInformation,
+                 mio::abm::LogDataForMovement>
+        historyPersonInf;
+    mio::History<mio::abm::TimeSeriesWriter, mio::abm::LogInfectionState> historyTimeSeries{
+        Eigen::Index(mio::abm::InfectionState::Count)};
+    mio::History<mio::abm::DataWriterToMemoryDelta, mio::abm::LogDataForMovement> historyPersonInfDelta;
+    auto sim = mio::abm::Simulation(mio::abm::TimePoint(0), std::move(world));
+    sim.advance(mio::abm::TimePoint(0) + mio::abm::hours(24), historyPersonInf, historyTimeSeries,
```

Comment:
Does this test depend on RNG? If so, please add mocks for all used distributions.
If not, could you add some comments what this test tries to do?

Reply:
No, all things tested are deterministic